### PR TITLE
Add initial unit tests

### PR DESCRIPTION
**Pull request by tnierman**
Adds some initial unit tests and the required frameworks to the `pkg/source/github` and `pkg/tool/ocm` packages. The unit tests have been implemented using [ginkgo](https://onsi.github.io/ginkgo/). 

As a part of these changes, two new types have been added to the `pkg/source/github` package:
- `testServer` objects mimic the responses expected by go-github from the v3 GitHub API. This allows us to instantiate test objects and return them as responses to the requests sent in our tests.
- `TestSource` serves as a drop-in replacement for the normal `Source` type, and is specifically designed for testing. They combine a `testServer` with a normal `Source` whose `baseURL` is pointed at the `testServer`. This allows us to test tools that have a GitHub source without actually sending requests to GitHub. 

Additionally, a few types have been added to the `pkg/utils/test` package:
- `dir` establishes a temporary directory for testing which can be disposed of at the end of each test. It contains the methods necessary to simulate a multitude of scenarios for the various tools we need to test
- `GithubEnv` combines a `dir` with a `TestSource` to create a complete GitHub testing environment. After initializing an environment with `NewGithubEnv()`, packages will then be able to build directory structures to simulate the scenario they want to test for, specify responses to various GitHub requests, and dispose of everything so a fresh environment can be recreated via a single object.

### CODE REVIEW

**Review by boranx — COMMENTED** [review]
hey @tnierman, 

Today I took a look at the PR, all looks good 👍  except one test, I'm not sure if this is due to my specs

```shell
• Failure [0.011 seconds]
Running Install():
/Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:37
  when linking the binary asset
  /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:496
    if a link is already present
    /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:497
      replaces it and links to the new binary [It]
      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:514

      Expected
          <string>: "/privat..."
      to equal        |
          <string>: "/var/fo..."

      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:528
 ```

**Review by boranx — COMMENTED** [review]
hey @tnierman, 

Today I took a look at the PR, all looks good 👍  except one test, I'm not sure if this is due to my specs

```shell
• Failure [0.011 seconds]
Running Install():
/Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:37
  when linking the binary asset
  /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:496
    if a link is already present
    /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:497
      replaces it and links to the new binary [It]
      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:514

      Expected
          <string>: "/privat..."
      to equal        |
          <string>: "/var/fo..."

      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:528
 ```

**Review by boranx — COMMENTED** [review]
hey @tnierman, 

Today I took a look at the PR, all looks good 👍  except one test, I'm not sure if this is due to my specs

```shell
• Failure [0.011 seconds]
Running Install():
/Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:37
  when linking the binary asset
  /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:496
    if a link is already present
    /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:497
      replaces it and links to the new binary [It]
      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:514

      Expected
          <string>: "/privat..."
      to equal        |
          <string>: "/var/fo..."

      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:528
 ```

**Review by boranx — COMMENTED** [review]
hey @tnierman, 

Today I took a look at the PR, all looks good 👍  except one test, I'm not sure if this is due to my specs

```shell
• Failure [0.011 seconds]
Running Install():
/Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:37
  when linking the binary asset
  /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:496
    if a link is already present
    /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:497
      replaces it and links to the new binary [It]
      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:514

      Expected
          <string>: "/privat..."
      to equal        |
          <string>: "/var/fo..."

      /Users/bseref/go/src/github.com/boranx/backplane-tools/pkg/tool/ocm/ocm_test.go:528
 ```

File: pkg/tool/ocm/ocm.go [hunk 1]
```diff
@@ -17,17 +17,21 @@ type Tool struct {
...
 }
 
+// Name returns the name this tool should be referenced by
 func (t *Tool) Name() string {
 	return "ocm"
 }
 
+// Install downloads this tool, verifies it's integrity via checksum, and links the newly downloaded binary
+// to the "latest" directory
 func (t *Tool) Install(rootDir, latestDir string) error {
```
Comment (geowa4) [inline pkg/tool/ocm/ocm.go:35]: I feel like this function could be broken up to make it more easily testable. For example, a function to return the `checksumAsset` and the `ocmBinaryAsset` from the `release` could be its own function. The code in this function is already broken down into checkpoints with comments; it looks like those could fairly easily become their own testable functions.

File: pkg/source/github/github_test.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+			// Create asset slice and tmp dir for the following tests
+			assets = []*gogithub.ReleaseAsset{asset1, asset2, asset3}
+
+			tmp, err = os.MkdirTemp(os.TempDir(), "")
+			Expect(err).ToNot(HaveOccurred())
+		})
+		// Clean the tmp dir and un-populate the assets slice after each test to avoid cross-contamination and memory leaks
+		// (Note: Changes to src are automatically handled at the Context level, so no direct cleanup required)
+		AfterEach(func() {
+			err = os.RemoveAll(tmp)
```
Comment (geowa4) [inline pkg/source/github/github_test.go:264]: Does this run even if a test panics? While I appreciate the need to test with a temp directory in such a file-centric application, I'm just a little concerned we have the possibility to leak temp directories over time. Could we push the IO dependency as far out on the fringe as we can to reduce the amount of test cases that need to do this? I'm not sure if this is possible, but if we can get the code to "this URL goes into a file at this location" and reuse that everywhere, we might be in a better place.

**Pull request closed without merging**